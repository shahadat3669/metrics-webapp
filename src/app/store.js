import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';

export const store = configureStore({
  reducer: {},
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
  devTools: process.env.NODE_ENV !== 'production',
});
