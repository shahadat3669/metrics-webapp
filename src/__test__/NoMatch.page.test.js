import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import NoMatch from '../pages/NoMatch';
import store from '../app/store';

describe('NoMatch', () => {
  test('renders NoMatch Page without crashing', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <NoMatch />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree.toJSON).toMatchSnapshot();
  });

  test('renders NoMatch Page without crashing', () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <NoMatch />
        </BrowserRouter>
      </Provider>,
    );
    container.querySelector('.mainTitle');
    expect(container.firstChild).toMatchSnapshot();

    container.querySelector('.first_color');
    expect(container.firstChild).toMatchSnapshot();
  });
});
