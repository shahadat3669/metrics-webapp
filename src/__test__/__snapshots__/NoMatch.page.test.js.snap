// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoMatch renders NoMatch Page without crashing 1`] = `undefined`;

exports[`NoMatch renders NoMatch Page without crashing 2`] = `
<div>
  <h2>
    Nothing to see here!
  </h2>
  <p>
    <a
      href="/"
    >
      Go to the home page
    </a>
  </p>
</div>
`;

exports[`NoMatch renders NoMatch Page without crashing 3`] = `
<div>
  <h2>
    Nothing to see here!
  </h2>
  <p>
    <a
      href="/"
    >
      Go to the home page
    </a>
  </p>
</div>
`;
